### PR TITLE
dev/core#2297 - E_NOTICE on contribution thankyou_date_time since time component no longer a separate field

### PR DESCRIPTION
**Pull request by demeritcowboy**
Overview
----------------------------------------
https://lab.civicrm.org/dev/core/-/issues/2297

E_NOTICE when entering a contribution with a thank you date (in the additional info section) since thankyou_date_time is no longer a submitted value for date fields.

Before
----------------------------------------
`Notice: Undefined index: thankyou_date_time in CRM_Contribute_Form_AdditionalInfo::postProcessCommon() (line 275 of .../web/sites/all/modules/civicrm/CRM/Contribute/Form/AdditionalInfo.php).`

After
----------------------------------------

Technical Details
----------------------------------------
Also looking back a bit at the history the check for isNull seems unnecessary since it doesn't seem to come in as the literal string 'null' anymore. Ditto setting the value to the literal string 'null' in the else clause.

Comments
----------------------------------------
Has test.

### CODE REVIEW

File: CRM/Contribute/Form/AdditionalInfo.php [hunk 1]
```diff
@@ -271,11 +271,11 @@ public static function postProcessCommon(&$params, &$formatted, &$form) {
       $formatted[$f] = $params[$f] ?? NULL;
     }
 
-    if (!empty($params['thankyou_date']) && !CRM_Utils_System::isNull($params['thankyou_date'])) {
-      $formatted['thankyou_date'] = CRM_Utils_Date::processDate($params['thankyou_date'], $params['thankyou_date_time']);
+    if (!empty($params['thankyou_date'])) {
+      $formatted['thankyou_date'] = CRM_Utils_Date::processDate($params['thankyou_date']);
```
Comment (eileenmcnaughton) [inline CRM/Contribute/Form/AdditionalInfo.php:275]: I don't think we need to do the processDate any more either - if you remove it & the test still passes it isn't needed

Comment (demeritcowboy) [reply CRM/Contribute/Form/AdditionalInfo.php:275]: The problem if I do that is that if you then just enter a time on the form with no date it crashes: `Incorrect datetime value: '4' for column thankyou_date`, where "4" is whatever the hour value is. It's an odd thing for someone to enter in this particular situation, but it lets you do it. So either processDate or a formRule or ?

Comment (eileenmcnaughton) [reply CRM/Contribute/Form/AdditionalInfo.php:275]: @demeritcowboy OK - we can leave that for another day then - this is definitely an improvment as is

**Comment by demeritcowboy** [comment]
Ok thanks.

**Pull request merged**